internal/services/compute/orchestrated_virtual_machine_scale_set.go

```diff
@@ -692,6 +692,28 @@ func OrchestratedVirtualMachineScaleSetTerminationNotificationSchema() *pluginsd
 	}
 }
 
+func OrchestratedVirtualMachineScaleSetPriorityMixPolicySchema() *pluginsdk.Schema {
+	return &pluginsdk.Schema{
+		Type:     pluginsdk.TypeList,
+		Optional: true,
+		MaxItems: 1,
+		Elem: &pluginsdk.Resource{
+			Schema: map[string]*pluginsdk.Schema{
+				"base_regular_priority_count": {
+					Type:     pluginsdk.TypeInt,
+					Optional: true,
+					Default:  0,
+				},
+				"regular_priority_percentage_above_base": {
+					Type:     pluginsdk.TypeInt,
+					Optional: true,
+					Default:  0,
+				},
```

Comment:
can we add validation for these two fields? What are the valid upper and lower bounds here? Something like:

```go
ValidateFunc: validation.IntBetween(0, 65535),
```

Reply:
Great point - thanks! 

We should definitely add this for the percentage (valid range is 0 - 100 since it's a percentage :)). 

As for the base number of regular VMs, In principle, the upper bound here is the same as the upper bound on the number of instances in the scale set (`sku.capacity` in the API and - looks like - `instances` in TF) as they both relate to the number of instances in a scale set; this is indeed how backend validation works. Now, it [looks like](https://github.com/hashicorp/terraform-provider-azurerm/blob/ddd9f54ac16a26eb1bff29c7f24b020a42541451/internal/services/compute/orchestrated_virtual_machine_scale_set_resource.go#L72C2-L77) the limit on `instances` is currently set to `1000`, so we should just go with that here too. That also is the publicly documented limit on VMSS Flex `sku.capacity`... 

(Having said that, I'll add that while the limit of 1000 instances is the current default limit, the VMSS team is looking to up this. So not sure if hardcoding this here is the best approach in general - should we really need a TF update if higher limits go GA? Nonetheless, that's orthogonal to this PR. Just adding for your awareness.)

Reply:
You're right, ideally, we don't want to have to do a minor update every time the VMSS limit is changed (let's say for example that they wanted to change it back). 

It's worth figuring out a way to sync the capacity limits with the VMSS Sku capacity and our own property. The most straightforward way seems like some sort of shared configuration in the REST API specs that represents the constraints on sku.capacity. Then priority mix in services such as TF could reference these constraints instead of using hardcoded values. For individual services, another approach is just to have some hardcoded constant to represent the max. 

I will reach out to VMSS team to find the best approach but don't want to add anything they are not made aware of.

![image](https://user-images.githubusercontent.com/43768045/222211894-b37762cf-7771-4d7d-98ff-1531bb61edde.png)